coda_v2_python_client/firebase_client_wrapper.py

```diff
@@ -111,3 +112,27 @@ def get_segment_count(self, dataset_id):
         if segment_count_doc is None:
             return None
         return segment_count_doc["segment_count"]
+
+    def get_message_ref(self, segment_id, message_id):
+        """ Gets Firestore database reference to a message.
+
+        :param segment_id: Id of a segment
+        :type segment_id: str
+        :param message_id: Id of a message
+        :type message_id: str
+        :return: A reference to a document in a Firestore database
+        :rtype: google.cloud.firestore_v1.document.DocumentReference
+        """
+        return self._client.document(f"datasets/{segment_id}/messages/{message_id}")
+
+    def get_message(self, dataset_id, message_id):
+        """Gets a message from a dataset by id.
+
+        :return: A message from a dataset.
+        :rtype: core_data_modules.data_models.message.Message | None
+        """
+        raw_message = self.get_message_ref(dataset_id, message_id).get().to_dict()
```

Comment:
Can you confirm you tested this on a multi-segment dataset? 

This doesn't look right as-is, as you are only checking the first segment. You actually need to check all of them until you find a match

Reply:
Sorry I didn't spot that. Thanks I've updated.